### ISSUE TITLE
oidcclient doesn't expand id attribute after 19.011

### ISSUE THREAD

**Issue by brutif** [OpenLiberty/open-liberty#10578]
in 19011, 
```
 <variable name="myclient" value="RP" />

    <openidConnectClient id="${myclient}" 
```
Would cause the id to be "RP" but beginning with 19.012 it is not being expanded, so the redirect URL incorrectly contains ${myclient}

**Comment by brutif** [comment]
Hi @brenthdaniel looking at trace, it looks to me like the kernel is calling setOidcClientConfig(ServiceReference<OidcClientConfig> ref) with the variable expanded in one case and not in the other. 

[trace_excerpt.txt](https://github.com/OpenLiberty/open-liberty/files/4110164/trace_excerpt.txt)
[server.xml.txt](https://github.com/OpenLiberty/open-liberty/files/4110179/server.xml.txt)

**Comment by mlynch7** [comment]
Hi @brutif @brenthdaniel Do we know what version of Liberty this will be fixed in? We are waiting on other issues to be resolved but cannot take them on while this is an issue (in particular https://github.com/OpenLiberty/open-liberty/issues/9516) Thanks

**Comment by brutif** [comment]
@mlynch7 we don't know yet, but we only have a few days to make 20.002. It would be 20.002 or 3.

**Comment by brenthdaniel** [comment]
Hi, 

 When we added a warning message stating that variables in ID values are not supported, we also made a change so that the value of the ID field remains consistent. Prior to 19.0.0.12, it existed in a strange state where it was interpreted as the variable string part of the time, and as the substituted value at other times. 

We can remove that change so that future versions behave the same as 19.0.0.11, but I strongly recommend that you avoid using variables in an ID value. In this case it doesn't matter quite so much because no other configurations reference the OIDC configuration, but in most cases it will cause problems. You will also continue to get a warning message because of this. 

I'll also open another issue for OIDC to consider adding another attribute that would override use of the ID attribute so that there would be a supported way to use a variable in this case.
